ruleguard/runner.go

```diff
@@ -103,9 +109,76 @@ func (rr *rulesRunner) run(f *ast.File) error {
 		})
 	}
 
+	if len(rr.rules.universal.commentRules) != 0 {
+		for _, commentGroup := range f.Comments {
+			for _, comment := range commentGroup.List {
+				rr.runCommentRules(comment)
+			}
+		}
+	}
+
 	return nil
 }
 
+func (rr *rulesRunner) runCommentRules(comment *ast.Comment) {
+	// We'll need that file to create a token.Pos from the artificial offset.
+	file := rr.ctx.Fset.File(comment.Pos())
+
+	for _, rule := range rr.rules.universal.commentRules {
+		var m commentMatchData
+		if rule.captureGroups {
+			result := rule.pat.FindStringSubmatchIndex(comment.Text)
+			if result == nil {
+				continue
+			}
+			for i, name := range rule.pat.SubexpNames() {
+				if i == 0 || name == "" {
+					continue
+				}
+				resultIndex := i * 2
+				beginPos := result[resultIndex+0]
+				endPos := result[resultIndex+1]
```

Comment:
looks like we're expecting `result` to be even, what about to add a check before loop with a panic? :)

Reply:
From the docs:

> If 'Index' is present, matches and submatches are identified by byte index pairs within the input string: result[2*n:2*n+1] identifies the indexes of the nth submatch.

https://golang.org/pkg/regexp/#Regexp.FindStringSubmatchIndex

I fixed another issue though. Checking for `-1` was too fragile.

> If an index is negative or text is nil, it means that subexpression did not match any string in the input.

So, it's safer to check for any negative number than for just `-1`.